src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/OtlpRetry.cs

```diff
@@ -157,16 +159,19 @@ private static int CalculateNextRetryDelay(int nextRetryDelayMilliseconds)
         return null;
     }
 
-    private static TimeSpan? TryGetHttpRetryDelay(HttpStatusCode statusCode, HttpResponseHeaders headers)
+    private static TimeSpan? TryGetHttpRetryDelay(HttpStatusCode? statusCode, HttpResponseHeaders? headers)
     {
-        Debug.Assert(headers != null, "headers was null");
+        if (!statusCode.HasValue)
+        {
+            return null;
+        }
 
 #if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
         return statusCode == HttpStatusCode.TooManyRequests || statusCode == HttpStatusCode.ServiceUnavailable
 #else
         return statusCode == (HttpStatusCode)429 || statusCode == HttpStatusCode.ServiceUnavailable
 #endif
-            ? headers.RetryAfter?.Delta
+            ? headers?.RetryAfter?.Delta
```

Comment:
Similar comment [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5394/files#r1505236151).

Are we now going to pass `null` for HttpHeaders?

Reply:
Yes, for network failures there will be no response headers.

Reply:
nit: Might help a bit with readability to rename `headers` -> `responseHeaders`